volume-group-operator/controllers/volumegroup_controller.go

```diff
@@ -53,14 +61,184 @@ type VolumeGroupReconciler struct {
 //+kubebuilder:rbac:groups="",resources=persistentvolumeclaims/finalizers,verbs=update
 
 func (r *VolumeGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := r.Log.WithValues("Request.Name", req.Name, "Request.Namespace", req.Namespace)
+
+	instance := &volumegroupv1.VolumeGroup{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+
+			logger.Info("VolumeGroup resource not found")
+
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	vgcObj, err := r.Utils.GetVolumeGroupClass(logger, *instance.Spec.VolumeGroupClassName)
+	if err != nil {
+		uErr := r.updateVolumeGroupStatusError(instance, logger, err.Error())
+		if uErr != nil {
+			logger.Error(uErr, "failed to update volumeGroup status", "VGName", instance.Name)
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	if r.DriverConfig.DriverName != vgcObj.Driver {
+		return ctrl.Result{}, nil
+	}
+
+	err = utils.ValidatePrefixedParameters(vgcObj.Parameters)
+	if err != nil {
+		logger.Error(err, "failed to validate parameters of volumegroupClass", "VGClassName", vgcObj.Name)
+		uErr := r.updateVolumeGroupStatusError(instance, logger, err.Error())
+		if uErr != nil {
+			logger.Error(uErr, "failed to update volumeGroup status", "VGName", instance.Name)
+		}
+
+		return ctrl.Result{}, err
+	}
+	parameters := utils.FilterPrefixedParameters(utils.VolumeGroupParameterPrefix, vgcObj.Parameters)
+
+	secretName := vgcObj.Parameters[utils.PrefixedVolumeGroupSecretNameKey]
+	secretNamespace := vgcObj.Parameters[utils.PrefixedVolumeGroupSecretNamespaceKey]
+	secret := make(map[string]string)
+	if secretName != "" && secretNamespace != "" {
+		secret, err = r.Utils.GetSecret(logger, secretName, secretNamespace)
+		if err != nil {
+			uErr := r.updateVolumeGroupStatusError(instance, logger, err.Error())
+			if uErr != nil {
+				logger.Error(uErr, "failed to update volumeGroup status", "VGName", instance.Name)
+			}
+
+			return reconcile.Result{}, err
+		}
+	}
+
+	if instance.GetDeletionTimestamp().IsZero() {
+		if err = r.Utils.AddFinalizerToVG(logger, instance); err != nil {
+			logger.Error(err, "Failed to add VolumeGroup finalizer")
+
+			return reconcile.Result{}, err
+		}
+
+	} else {
+		if utils.Contains(instance.GetFinalizers(), utils.VolumeGroupFinalizer) {
+			volumeGroupContent, uErr := r.Utils.GetVolumeGroupContent(logger, instance)
+			if uErr != nil {
+				if !errors.IsNotFound(uErr) {
+					logger.Error(uErr, "failed to get volumeGroupContent", *instance.Spec.Source.VolumeGroupContentName)
```

Comment:
IMO this log should be in `GetVolumeGroupContent` function, (there are other logs in the reconcile that can be handled in their functions and shouldn't be handled in the reconcile)

Reply:
done